packages/server/src/common/features/contextactions/smart-connector-item-provider.ts

```diff
@@ -0,0 +1,182 @@
+     *  the edge to a new node when the source node has the ID of the key
+     *  otherwise, the default edge will be used */
+    protected edgeTypes: Record<string, string | undefined>;
+
+    getItems(args?: Args): SmartConnectorGroupItem[] {
+        const handlers = this.operationHandlerRegistry.getAll().filter(CreateOperationHandler.is) as CreateOperationHandler[];
+        this.counter = 0;
+        const nodes = this.createSmartConnectorGroupItem(handlers, CreateNodeOperation.KIND, args?.nodeType as string,
+            this.smartConnectorNodeSettings.showOnlyForChildren);
+        const edges = this.createSmartConnectorGroupItem(handlers, CreateEdgeOperation.KIND, args?.nodeType as string,
+            this.smartConnectorEdgeSettings.showOnlyForChildren);
+        return [
+            {
+                id: 'smart-connector-node-group',
+                label: 'Nodes',
+                actions: [],
+                children: nodes,
+                icon: 'symbol-property',
+                sortString: 'A',
+                ...this.smartConnectorNodeSettings
+            },
+            {
+                id: 'smart-connector-edge-group',
+                label: 'Edges',
+                actions: [],
+                children: edges,
+                icon: 'symbol-property',
+                sortString: 'B',
+                ...this.smartConnectorEdgeSettings
+            }
+        ];
+    }
+
+    createSmartConnectorGroupItem(handlers: CreateOperationHandler[], kind: string, selectedNodeType: string,
+            showOnly?: SmartConnectorGroupUIType): PaletteItem[] {
+        const includedInNodeFilter = (e: string): boolean => !!this.nodeOperationFilter[selectedNodeType]?.includes(e);
+        const paletteItems = handlers
+            .filter(handler => handler.operationType === kind && (selectedNodeType &&
+                this.nodeOperationFilter[selectedNodeType] ? !handler.elementTypeIds.some(includedInNodeFilter) : true))
+            .map(handler => handler.getTriggerActions().map(action => this.create(action, handler.label, selectedNodeType)))
+            .reduce((accumulator, value) => accumulator.concat(value), [])
+            .sort((a, b) => a.sortString.localeCompare(b.sortString));
+        if (showOnly === SmartConnectorGroupUIType.Icons) {
+            if (paletteItems.every(paletteItem => paletteItem.icon !== '')) {
+                this.logger.warn('Not all elements have icons. Labels will be shown, check settings for smart connector.')
+                return paletteItems;
+            }
+            paletteItems.forEach(paletteItem => paletteItem.label = '');
+        }
+        else if (showOnly === SmartConnectorGroupUIType.Labels) {
+            paletteItems.forEach(paletteItem => paletteItem.icon = '');
+        }
+        return paletteItems;
+    }
+
+    create(action: PaletteItem.TriggerElementCreationAction, label: string, nodeType: string): PaletteItem | SmartConnectorNodeItem {
```

Comment:
```suggestion
    protected create***(action: PaletteItem.TriggerElementCreationAction, label: string, nodeType: string): PaletteItem | SmartConnectorNodeItem {
```

Please provide for * a specific name, what are you creating here? `createSmartConnectorItem` probably?